File: medbrains/apps/ios/MedBrainsStaff/Features/StaffModules.swift
import MedBrainsKit
import SwiftUI

/// The staff registry, in the order apps/mobile-staff/src/modules/index.ts
/// keeps it. Order is not cosmetic: the shell opens on the first permitted
/// module, so this list decides where each role lands. A receptionist holds
/// `billing.invoices.list` for taking payments, and with reception at the
/// bottom that landed them in Billing every morning. Desk and clinical
/// modules first, cross-cutting ones after.
enum StaffModules {
    static let registry: [AppModule] = [
        module("doctor", "Doctor", "stethoscope", ["opd.visit.update"], apps: ["Mobile-Doctor"], phase: 3),
        module("nurse", "Nurse", "cross.case", ["nurse.dashboard.view"], apps: ["Mobile-Nurse"], phase: 2),
        module("reception", "Reception", "person.badge.clock", ["front_office.queue.list"], apps: ["Desktop-Kiosk", "Desktop-Workstation", "Mobile-Admin", "Mobile-Security"], phase: 5),
        module("pharmacy", "Pharmacy", "pills", ["pharmacy.prescriptions.list"], apps: ["Mobile-Pharmacist"], phase: 5),
        module("lab", "Lab", "testtube.2", ["lab.orders.list"], apps: ["Mobile-LabTech", "Mobile-Phlebo"], phase: 5),
        module("blood-bank", "Blood Bank", "drop", ["blood_bank.inventory.list"], apps: ["Mobile-LabTech", "Mobile-Nurse", "Mobile-Doctor"], phase: 5),
        module("billing", "Billing", "indianrupeesign.circle", ["billing.invoices.list"], apps: ["Mobile-Admin", "Desktop-Workstation"], phase: 5),
        module("bme", "BME", "wrench.and.screwdriver", ["bme.equipment.list"], apps: ["Mobile-BME", "Desktop-DeviceBridge", "Mobile-Vendor"], phase: 5),
        module("facilities", "Facilities", "building.2", ["facilities.work_orders.list"], apps: ["Mobile-BME", "Mobile-Housekeeping", "TV-Wayfinding"], phase: 5),
        module("housekeeping", "Housekeeping", "bed.double", ["housekeeping.cleaning.list"], apps: ["Mobile-Housekeeping", "TV-Ward"], phase: 5),
        module("security", "Security", "shield.lefthalf.filled", ["security.incidents.list"], apps: ["Mobile-Security", "TV-Emergency"], phase: 5),
        module("hr", "HR", "person.2", ["hr.attendance.list"], apps: ["Mobile-Admin"], phase: 5),
        module("device-sync", "Device sync", "arrow.triangle.2.circlepath", [], apps: ["Mobile-Admin"], phase: 6),
    ]

    private static func module(_ id: String, _ name: String, _ symbol: String, _ all: [String], apps: [String], phase: Int) -> AppModule {
        AppModule(id: id, displayName: name, symbol: symbol, requiredPermissions: all, appCodes: apps) {
            AnyView(PlaceholderModuleHome(module: AppModule(id: id, displayName: name, symbol: symbol, requiredPermissions: all, appCodes: apps) { AnyView(EmptyView()) }, phase: phase))
        }
    }
}

File: medbrains/apps/android/app-staff/src/main/kotlin/com/medbrains/staff/StaffModules.kt
package com.medbrains.staff

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccountBalanceWallet
import androidx.compose.material.icons.filled.Apartment
import androidx.compose.material.icons.filled.Bloodtype
import androidx.compose.material.icons.filled.Build
import androidx.compose.material.icons.filled.CleaningServices
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material.icons.filled.LocalHospital
import androidx.compose.material.icons.filled.Medication
import androidx.compose.material.icons.filled.MonitorHeart
import androidx.compose.material.icons.filled.Science
import androidx.compose.material.icons.filled.Security
import androidx.compose.material.icons.filled.SupportAgent
import androidx.compose.material.icons.filled.Sync
import androidx.compose.ui.graphics.vector.ImageVector
import com.medbrains.kit.AppModule
import com.medbrains.staff.doctor.DoctorModule
import com.medbrains.staff.nurse.NurseModule

/**
 * The staff registry, in the order apps/mobile-staff/src/modules/index.ts
 * keeps it. Order is not cosmetic: the shell opens on the first permitted
 * module, so this list decides where each role lands. Desk and clinical
 * modules first, cross-cutting ones after.
 */
object StaffModules {
    val registry: List<AppModule> = listOf(
        AppModule(id = "doctor", displayName = "Doctor", requiredPermissions = listOf("opd.visit.update"), appCodes = listOf("Mobile-Doctor"), icon = Icons.Filled.MonitorHeart) { DoctorModule() },
        AppModule(id = "nurse", displayName = "Nurse", requiredPermissions = listOf("nurse.dashboard.view"), appCodes = listOf("Mobile-Nurse"), icon = Icons.Filled.LocalHospital) { NurseModule() },
        module("reception", "Reception", Icons.Filled.SupportAgent, listOf("front_office.queue.list"), listOf("Desktop-Kiosk", "Desktop-Workstation", "Mobile-Admin", "Mobile-Security"), 5),
        module("pharmacy", "Pharmacy", Icons.Filled.Medication, listOf("pharmacy.prescriptions.list"), listOf("Mobile-Pharmacist"), 5),
        module("lab", "Lab", Icons.Filled.Science, listOf("lab.orders.list"), listOf("Mobile-LabTech", "Mobile-Phlebo"), 5),
        module("blood-bank", "Blood Bank", Icons.Filled.Bloodtype, listOf("blood_bank.inventory.list"), listOf("Mobile-LabTech", "Mobile-Nurse", "Mobile-Doctor"), 5),
        module("billing", "Billing", Icons.Filled.AccountBalanceWallet, listOf("billing.invoices.list"), listOf("Mobile-Admin", "Desktop-Workstation"), 5),
        module("bme", "BME", Icons.Filled.Build, listOf("bme.equipment.list"), listOf("Mobile-BME", "Desktop-DeviceBridge", "Mobile-Vendor"), 5),
        module("facilities", "Facilities", Icons.Filled.Apartment, listOf("facilities.work_orders.list"), listOf("Mobile-BME", "Mobile-Housekeeping", "TV-Wayfinding"), 5),
        module("housekeeping", "Housekeeping", Icons.Filled.CleaningServices, listOf("housekeeping.cleaning.list"), listOf("Mobile-Housekeeping", "TV-Ward"), 5),
        module("security", "Security", Icons.Filled.Security, listOf("security.incidents.list"), listOf("Mobile-Security", "TV-Emergency"), 5),
        module("hr", "HR", Icons.Filled.Groups, listOf("hr.attendance.list"), listOf("Mobile-Admin"), 5),
        // Phase 6 defines this module's own permission; until then the pairing roster gates the placeholder.
        module("device-sync", "Device sync", Icons.Filled.Sync, listOf("devices.pairing.paired_list"), listOf("Mobile-Admin"), 6),
    )

    private fun module(id: String, name: String, icon: ImageVector, all: List<String>, apps: List<String>, phase: Int): AppModule {
        lateinit var self: AppModule
        self = AppModule(id = id, displayName = name, requiredPermissions = all, appCodes = apps, icon = icon) {
            PlaceholderModuleHome(self, phase)
        }
        return self
    }
}
